src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/IQueueItem.cs

```diff
@@ -37,6 +38,12 @@ internal interface IQueueItem<TRequestContext>
     /// </summary>
     Task<(TRequestContext, TRequest)?> CreateRequestContextAsync<TRequest>(IMethodHandler handler, RequestHandlerMetadata requestHandlerMetadata, AbstractLanguageServer<TRequestContext> languageServer, CancellationToken cancellationToken);
 
+    /// <summary>
+    /// Handles when the queue needs to manually fail a request before the
+    /// handler is invoked without shutting down the entire queue.
+    /// </summary>
+    void FailRequest(Exception ex);
```

Comment:
Perhaps take a message?

Reply:
the exception message is already logged, so I think fine as-is.  we also log more details before the exception hits

Reply:
nvm I see what you mean.  I will try